xds/pkg/server/server.go

```diff
@@ -33,40 +38,99 @@ const (
 	grpcMaxConcurrentStreams = 1000000
 )
 
+// HealthCheck implements the health check logic exposed by the server.
+// It is implemented by internal components of the server and returns
+// an error if the component is unhealthy.
+type HealthCheck func(ctx context.Context) error
```

Comment:
Just a thought: I've used https://github.com/heptiolabs/healthcheck in the past (not that it's been updated in a while) to build out composable health checks. Just a thought that it might save some work building this out ourselves.

But it's also easy enough to do that it's not a huge deal, just thought I'd throw the idea out there.

Reply:
Yeah I think its simple enough in this case to not drag the dependency